src/mcode-pilot/components/TreatmentOptionsOutcomesIcons/TreatmentOptionsOutcomesIcons.js

```diff
@@ -11,142 +11,53 @@ export default class TreatmentOptionsOutcomesIcons extends Component {
-        } else {
-            index = comparedTreatmentData.findIndex(el => el.displayName === treatment);
-            totalNumSurvive = comparedTreatmentData[index][timescale];
-            totalPatients = comparedTreatmentData[index].totalPatients;
-        }
-
-        const numSurvive = Math.floor(totalNumSurvive / totalPatients * 100);
-        return { numSurvive, index };
+    selectIconsTreatment = treatmentName => {
+        const { similarPatientTreatmentsData } = this.props;
+        const treatment = similarPatientTreatmentsData.filter(treatmentData => treatmentData.displayName === treatmentName)[0];
+        this.setState({ selectedTreatment: treatment });
     }
 
     render() {
-        const { includedTreatmentData, comparedTreatmentData, timescaleToggle } = this.props;
-        const { selectedTreatmentType, selectedTreatment } = this.state;
-
-        if (includedTreatmentData.length === 0) {
-            return <div className="treatment-options-outcomes-icons note">No included treatment chosen.</div>;
-        }
-
+        const { similarPatientTreatmentsData, timescaleToggle } = this.props;
+        const { selectedTreatment } = this.state;
         const survivalMap = { 1: 'oneYrSurvival', 3: 'threeYrSurvival', 5: 'fiveYrSurvival' };
-        const includedTreatmentName = includedTreatmentData[0].displayName;
-        const treatment = selectedTreatment ? selectedTreatment : includedTreatmentName;
-        const combinedTreatmentNames = _.map(comparedTreatmentData, 'displayName');
-        const { numSurvive: includedNumSurvive } = this.getNumSurvive(includedTreatmentName, survivalMap[timescaleToggle]);
-        const { numSurvive: selectedNumSurvive, index: selectedIndex } = this.getNumSurvive(treatment, survivalMap[timescaleToggle]);
-        const additionalNumSurvive = selectedNumSurvive - includedNumSurvive;
-
-        let normalizedNumSideEffects = 0;
-        let normalizedSideEffects = [];
-        if (selectedIndex >= 0) {
-            const totalReporting = comparedTreatmentData[selectedIndex].sideEffects.totalReporting;
-            const totalPatients = comparedTreatmentData[selectedIndex].totalPatients;
-            normalizedNumSideEffects = Math.round(totalReporting / totalPatients * 100);
-
-            const effects = comparedTreatmentData[selectedIndex].sideEffects.effects;
-            Object.keys(effects).forEach(key => {
-                normalizedSideEffects.push({
-                    name: key,
-                    numEffects: Math.round(effects[key] / totalPatients * 100)
-                });
-            });
-            normalizedSideEffects.sort((a, b) => b.numEffects - a.numEffects);
-        }
+        const timescale = survivalMap[timescaleToggle];
+        const similarPatientTreatmentNames = _.map(similarPatientTreatmentsData, 'displayName');
+        const numSurvive = Math.floor(selectedTreatment[timescale] / selectedTreatment.totalPatients * 100);
```

Comment:
[KG1] If no treatment is "selected", or in this case if you select enough filters such that no patients meet the criteria and try switch to icon view, `selectedTreatment` is undefined and the error crashes the app.